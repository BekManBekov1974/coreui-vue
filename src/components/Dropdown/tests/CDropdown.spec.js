import { mount } from '@vue/test-utils'
import Component from '../CDropdown'

const ComponentName = 'CDropdown'
const defaultWrapper = mount(Component)
const customWrapper = mount(Component, {
  propsData: {
    togglerText: 'Dropdown button',
    show: true,
    disabled: true,
    addMenuClasses: 'additional-menu-class',
    addTogglerClasses: 'additional-toggler-class',
    nav: false,
    noCaret: true,
    variant: 'success',
    size: 'lg',
    split: true,
    offset: 20,
    placement: 'bottom-end',
    noFlip: true,
  },
  slots: {
    default: 'CDropdown subcomponents'
  }
})

describe(ComponentName, () => {
  it('has a name', () => {
    expect(Component.name).toMatch(ComponentName)
  })
  it('renders correctly', () => {
    expect(defaultWrapper.element).toMatchSnapshot()
  })
  it('renders correctly', () => {
    expect(customWrapper.element).toMatchSnapshot()
  })
})
