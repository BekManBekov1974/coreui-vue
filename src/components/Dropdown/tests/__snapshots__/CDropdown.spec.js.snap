// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CDropdown renders correctly 1`] = `
<div
  class="c-dropdown"
>
  <button
    aria-expanded="false"
    aria-haspopup="true"
    class="c-btn c-dropdown-toggle"
  >
    Dropdown
  </button>
   
  <!---->
   
  <div
    class="c-dropdown-menu"
  />
</div>
`;

exports[`CDropdown renders correctly 2`] = `
<div
  class="c-dropdown c-show c-btn-group"
>
  <button
    class="additional-toggler-class c-btn c-btn-lg c-disabled c-btn-success"
  >
    Dropdown button
  </button>
   
  <button
    aria-expanded="true"
    aria-haspopup="true"
    class="c-dropdown-toggle c-dropdown-toggle-split additional-toggler-class c-btn c-btn-lg c-disabled c-btn-success"
  />
   
  <div
    class="additional-menu-class c-dropdown-menu c-show"
  >
    CDropdown subcomponents
  </div>
</div>
`;
